src/container.h

```diff
@@ -86,6 +86,9 @@ class RingDeque {
     read_pos_ = (read_pos_ + 1) % CAPACITY;
   }
 
+  unsigned front_index() const { return read_pos_; }
```

Comment:
Exposing these values breaks encapsulation of this RingDeque; we should avoid it.

You do this to be able to follow the lower level queue and then trying to keep track of the index and attempt to re-create that situation with the deque.

Don't do it that complicated: after all, the queue with the underlying PRU is handled 'manually' anyway: we always know which index we entered something when we stored an element in there. So just let the `MotionQueue::Enqueue(MotionSegment *element)` return an integer which represents the index in which it was stored.

Then things suddenly get simple and understandable: the index in the StatusRegister suddenly makes sense, as it has a 1:1 correlation with the index returned by the Enqueue(); your history queue can just be implemented by simply allocating an array of of QUEUE_SIZE with the elements in there.